.github/workflows/release.yml

```diff
@@ -0,0 +1,264 @@
+          label <- grep(">N Parameters<", html_txt, fixed = T)
+          diff_label <- diff(label)
+          to_rm <- which(diff_label == 5)
+          to_rm_lines <- label[to_rm+1]
+          for (i in to_rm_lines) {
+            start <- grep("<tr", html_txt, fixed = TRUE)
+            end <- grep("/tr", html_txt, fixed = TRUE)
+            start <- max(start[start %in% (i-2):(i-1)])
+            end <- min(end[end %in% (i+1):(i+2)])
+            if(length(start) == 1 & length(end) == 1) {
+              html_txt[start:end] <- ""
+            }
+          }
+          # Add title to table of contents
+          where_add <- grep('<nav id="TOC" role="doc-toc">',
+                            html_txt, fixed = TRUE)
+          html_txt <- append(html_txt,
+                             "<h1>Table of Contents</h1>",
+                             after = where_add[1])
+          # Add title to references
+          where_add <- (grep("references csl-bib-body hanging-indent",
+                             html_txt, fixed = TRUE) - 1)
+          html_txt <- append(html_txt,
+            '<h1 data-number="17" id="sec:references"><span class="header-section-number">17</span> References</h1>',
+            after = where_add)
+          # Add references to table of contents
+          where_add <- grep("</nav>", html_txt, fixed = TRUE)-2
+          html_txt <- append(html_txt,
+            '<li><a href="#sec:references"><span class="toc-section-number">17</span> References</a></li>',
+            after = where_add)
+          writeLines(html_txt, "SS330_User_Manual_release.html")
+          writeLines(html_txt, "docs/SS330_User_Manual_release.html")
+        shell: Rscript {0}     
+
+      - name: Commit html of manual to docs folder
+        uses: EndBug/add-and-commit@v9
+        with: 
+          add: './docs/SS330_User_Manual_release.html'
+          message: 'update manual html in docs'
+
+# PDF Portion
+      - name: remove rendered file, check that is gone.
+        run: |
+          rm -rf SS330_User_Manual.pdf
+          ls
+     # - name: add lua filter to the repository
+     #   run: wget https://raw.githubusercontent.com/klpn/accpdf/master/common/tagged-filter.lua      
+        
+      - name: Compile LaTeX document
+        uses: xu-cheng/latex-action@v2
+        with:
+          root_file: SS330_User_Manual.tex
+          #args: --lua-filter=tagged-filter.lua
+          latexmk_use_lualatex: true
+
+# Change name of pdf file to include "_release"
+      - name: Add _release to pdf file name
+        run: |
+          mv SS330_User_Manual.pdf SS330_User_Manual_release.pdf
+
+# Update the version in index.md      
+      - name: Update version in index.md
+        run: |
+          version <- readLines("version.txt", encoding = "UTF-8")
+          index <- readLines("docs/index.md", encoding = "UTF-8")
+          version <- gsub("v", "", version)
+          index[6] <- gsub("[0-9].[0-9][0-9].[0-9][0-9]", version, index[6])
```

Comment:
`[0-9][0-9]` can be rewritten to `[0-9]{2}` if you want

Reply:
It ends up having to be index <- gsub("v[0-9]\\.[0-9]{2}\\.[0-9]{2}", version, index) for some reason. When i tried without the second {2} with version <- "v.3.30.22" it returned in the index "v.3.30.221".

Reply:
Sorry for the mistake in the code that I wrote, you are right. It has to be that way otherwise it will only replace the first numeric value after the last period. So the trailing 1 in "v.3.30.221" is left over from the previous version's last number being attached to the new version.

Reply:
Yeah no worries! I appreciate your suggestions! 😃